crates/core/src/store.rs

```diff
@@ -366,14 +389,14 @@ impl StoreBuilder {
         self.build_with_data(T::default())
     }
 
-    fn with_wasi(&mut self, f: impl FnOnce(&mut Wasi)) {
+    fn with_wasi(&mut self, f: impl FnOnce(&mut WasiCtxBuilder)) {
         let _ = self.try_with_wasi(|wasi| {
             f(wasi);
             Ok(())
         });
     }
 
-    fn try_with_wasi(&mut self, f: impl FnOnce(&mut Wasi) -> Result<()>) -> Result<()> {
+    fn try_with_wasi(&mut self, f: impl FnOnce(&mut WasiCtxBuilder) -> Result<()>) -> Result<()> {
```

Comment:
Might be nice to update these to take two closures e.g.

```suggestion
    fn try_with_wasi(
      &mut self,
      preview1: impl FnOnce(&mut WasiCtxBuilder) -> Result<()>,
      preview2: impl FnOnce(WasiCtxBuilder) -> Result<WasiCtxBuilder>)
    ) -> Result<()> {
```

Which would let you avoid the `take` juggling in all the methods above. :shrug:

Reply:
I tried this out because it seemed like a good idea, but it turns out to require even more code due to the fact that some methods can be implemented using a closure based approach (due to their arguments not being moveable into two separate closures). Here's what the code looks like: https://github.com/fermyon/spin/compare/upgrade-wasmtime-10-0...try-suggestion